cmd/arduino-flasher-cli/daemon/daemon.go

```diff
@@ -0,0 +1,124 @@
+// This file is part of arduino-flasher-cli.
+//
+// Copyright 2025 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-flasher-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/arduino/arduino-flasher-cli/cmd/feedback"
+	"github.com/arduino/arduino-flasher-cli/cmd/i18n"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	rpc "github.com/arduino/arduino-flasher-cli/rpc/cc/arduino/flasher/cli/commands/v1"
+)
+
+func NewDaemonCommand(srv rpc.ArduinoCoreServiceServer) *cobra.Command {
+	var daemonPort string
+	var maxGRPCRecvMsgSize int
+	daemonCommand := &cobra.Command{
+		Use:     "daemon",
+		Short:   i18n.Tr("Run the Arduino Flasher CLI as a gRPC daemon."),
+		Example: "  " + os.Args[0] + " daemon",
+		Args:    cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			runDaemonCommand(srv, daemonPort, maxGRPCRecvMsgSize)
+		},
+	}
+
+	daemonCommand.Flags().StringVar(&daemonPort,
+		"port", "50051",

```

Comment:
Didn't we use the same port in the Arduino CLI? I would change it to avoid possible clashes.
```suggestion
		"port", "50052",
```